blackmarket/Networking/Services/AuthenticationServices.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  AuthenticationServices.swift
+//  ios-base
+//
+//  Created by Germán Stábile on 6/8/20.
+//  Copyright © 2020 Rootstrap Inc. All rights reserved.
+//
+
+import RSSwiftNetworking
+
+internal class AuthenticationServices {
+    
+    enum AuthError: Error {
+        case userSessionInvalid
+    }
+    
+    // MARK: - Properties
+        
+    private let sessionManager: SessionManager
+    
+    private let apiClient: APIClient
+    
+    init(
+        sessionManager: SessionManager = .shared,
+        apiClient: APIClient = Client.shared
+    ) {
+        self.sessionManager = sessionManager
+        self.apiClient = apiClient
+    }
+    
+    func login(
+        email: String,
+        password: String,
+        completion: @escaping (Result<Void, Error>) -> Void
```

Comment:
I think we should start using Swift Concurrency already. Instead of dragging all the tech debt from the ios-base and then need to update all the codebases

Reply:
agree, you can check in swiftui-base if you want